sdk/python/kfp/v2/dsl/artifact.py

```diff
@@ -0,0 +1,332 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Base class for MLMD artifact ontology in KFP SDK."""
+from typing import Any, Dict, Optional
+
+from absl import logging
+import enum
+import importlib
+from google.protobuf import json_format
+import yaml
+
+from kfp.pipeline_spec import pipeline_spec_pb2
+
+_KFP_ARTIFACT_TITLE_PATTERN = 'kfp.{}'
+_KFP_ARTIFACT_ONTOLOGY_MODULE = 'kfp.v2.dsl.artifacts'
+
+
+# Enum for property types.
+# This is introduced to decouple the MLMD ontology with Python built-in types.
+class PropertyType(enum.Enum):
+  INT = 1
+  DOUBLE = 2
+  STRING = 3
+
+
+class Property(object):
+  """Property specified for an Artifact."""
+
+  # Mapping from Python enum to primitive type in the IR proto.
+  _ALLOWED_MLMD_TYPES = {
+      PropertyType.INT: pipeline_spec_pb2.PrimitiveType.INT,
+      PropertyType.DOUBLE: pipeline_spec_pb2.PrimitiveType.DOUBLE,
+      PropertyType.STRING: pipeline_spec_pb2.PrimitiveType.STRING,
+  }
+
+  def __init__(self, type, description: str = None):
```

Comment:
nit: `type: PropertyType` and `description: Optional[str] = None`

Reply:
Done